pkg/apis/duck/v1beta1/delivery_conversion_test.go

```diff
@@ -44,3 +50,233 @@ func TestDeliveryStatusConversionBadType(t *testing.T) {
+	var backoffPolicy v1.BackoffPolicyType = v1.BackoffPolicyLinear
+	var backoffPolicyExp v1.BackoffPolicyType = v1.BackoffPolicyExponential
+	var backoffPolicyBad v1.BackoffPolicyType = "garbage"
+	badPolicyString := `unknown BackoffPolicy, got: "garbage"`
+
+	tests := []struct {
+		name string
+		in   *v1.DeliverySpec
+		err  *string
+	}{{
+		name: "min configuration",
+		in: &v1.DeliverySpec{
+			DeadLetterSink: &pkgduck.Destination{
+				URI: apis.HTTP("example.com"),
+			},
+		},
+	}, {
+		name: "with retry",
+		in: &v1.DeliverySpec{
+			Retry: &retryCount,
+			DeadLetterSink: &pkgduck.Destination{
+				URI: apis.HTTP("example.com"),
+			},
+		},
+	}, {
+		name: "with linear backoff",
+		in: &v1.DeliverySpec{
+			Retry:         &retryCount,
+			BackoffPolicy: &backoffPolicy,
+			DeadLetterSink: &pkgduck.Destination{
+				URI: apis.HTTP("example.com"),
+			},
+		},
+	}, {
+		name: "with exp backoff",
+		in: &v1.DeliverySpec{
+			Retry:         &retryCount,
+			BackoffPolicy: &backoffPolicyExp,
+			DeadLetterSink: &pkgduck.Destination{
+				URI: apis.HTTP("example.com"),
+			},
+		},
+	}, {
+		name: "with bad backoff",
+		in: &v1.DeliverySpec{
+			Retry:         &retryCount,
+			BackoffPolicy: &backoffPolicyBad,
+			DeadLetterSink: &pkgduck.Destination{
+				URI: apis.HTTP("example.com"),
+			},
+		},
+		err: &badPolicyString,
+	}}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ver := &DeliverySpec{}
+			err := ver.ConvertFrom(context.Background(), test.in)
+			if err != nil {
+				if test.err == nil || *test.err != err.Error() {
+					t.Errorf("ConvertFrom() = %v", err)
+				}
+				return
+			}
+			got := &v1.DeliverySpec{}
+			if err := ver.ConvertTo(context.Background(), got); err != nil {
+				t.Errorf("ConvertTo() = %v", err)
+			}
+			if diff := cmp.Diff(test.in, got); diff != "" {
+				t.Errorf("roundtrip (-want, +got) = %v", diff)
+			}
+		})
+	}
+}
+
+// Test v1beta1 -> v1 -> v1beta1
+func TestDeliveryStatuscConversion(t *testing.T) {
+	tests := []struct {
+		name string
+		in   *DeliveryStatus
+		err  *string
+	}{{
+		name: "min configuration",
+		in: &DeliveryStatus{
+			DeadLetterChannel: &duckv1.KReference{
+				Kind:       "dlKind",
+				Namespace:  "dlNamespace",
+				Name:       "dlName",
+				APIVersion: "dlAPIVersion",
+			},
+		},
+	}}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ver := &v1.DeliveryStatus{}
+			err := test.in.ConvertTo(context.Background(), ver)
+			if err != nil {
+				if test.err == nil || *test.err != err.Error() {
+					t.Errorf("ConvertTo() = %v", err)
+				}
+				return
+			}
+			got := &DeliveryStatus{}
+			if err := got.ConvertFrom(context.Background(), ver); err != nil {
+				t.Errorf("ConvertFrom() = %v", err)
+			}
+			if diff := cmp.Diff(test.in, got); diff != "" {
+				t.Errorf("roundtrip (-want, +got) = %v", diff)
+			}
+		})
+	}
+}
+
+// Test v1 -> v1beta1 -> v1
+func TestDeliveryStatuscConversionV1(t *testing.T) {
```

Comment:
Typo in func name 

Also to confirm you wrote this test so that avoid side-effects failure on the input?

Reply:
No, that was covered by the RT tests. I added these because we really didn't have any tests, so wanted to increase coverage.